Bottom: ce994ec4d4ef56378d9d534ae78acc5e588ff757
Top:    514ceabe6f316d98979192aa5bf8042dc32875d9
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-13 15:53:16 -0500

haproxy configuration


---
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
index f908c8f03..b73f26e20 100644
--- a/ansible-playbook/group_vars/docker-hosts/networking.yml
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -1,30 +1,20 @@
 ---
 
 svn_hostname: svn.novatech-llc.com
-svn_port_args:
-- "{{ svn_ip_addr }}:80:80"
-- "{{ svn_ip_addr }}:443:443"
+svn_port_args: []
 
 gitlab_hostname: git.novatech-llc.com
 gitlab_port_args:
-- "{{ gitlab_ip_addr }}:80:80"
-- "{{ gitlab_ip_addr }}:443:443"
 - "{{ gitlab_ip_addr }}:22:22"
 
 wiki_hostname: wiki.novatech-llc.com
-wiki_port_args:
-- "{{ wiki_ip_addr }}:80:80"
-- "{{ wiki_ip_addr }}:443:443"
+wiki_port_args: []
 
 mantisbt_hostname: mantis.novatech-llc.com
-mantisbt_port_args:
-- "{{ mantisbt_ip_addr }}:80:80"
-- "{{ mantisbt_ip_addr }}:443:443"
+mantisbt_port_args: []
 
 build_hostname: buildsystem.novatech-llc.com
-build_port_args:
-- "{{ build_ip_addr }}:80:80"
-- "{{ build_ip_addr }}:443:443"
+build_port_args: []
 
 testssh_hostname: testssh.novatech-llc.com
 testssh_port_args:
@@ -32,9 +22,7 @@ testssh_port_args:
 
 exim4_hostname: mail.novatech-llc.com
 exim4_port_args:
-- "{{ exim4_ip_addr }}:25:25"
-- "{{ exim4_ip_addr }}:465:465"
-- "{{ exim4_ip_addr }}:587:587"
+- "25"
 
 openldap_hostname: ldap.novatech-llc.com
 openldap_port_args:
@@ -42,8 +30,13 @@ openldap_port_args:
 - "{{ openldap_ip_addr }}:636:636"
 
 phpldapadmin_hostname: ldap.novatech-llc.com
-phpldapadmin_port_args:
-- "{{ phpldapadmin_ip_addr }}:80:80"
-- "{{ phpldapadmin_ip_addr }}:443:443"
+phpldapadmin_port_args: []
 
 phpmyadmin_hostname: phpmyadmin.novatech-llc.com
+
+buildbot_hostname: buildbot.novatech-llc.com
+buildbot_port_args:
+- "{{ buildbot_ip_addr }}:9989:9989"
+
+nexus_hostname: nexus.novatech-llc.com
+nexus_port_args: []
diff --git a/portmap.txt b/portmap.txt
new file mode 100644
index 000000000..6318bf53e
--- /dev/null
+++ b/portmap.txt
@@ -0,0 +1,118 @@
+              *:111
+              *:9101
+              *:9103
+             :::111
+            ::1:25
+127.000.000.001:25      smtp
+127.000.000.001:9102    bacula-fd
+
+172.016.000.100:22      ssh
+172.016.000.100:3306    mysql
+
+svn.novatech-llc.com
+172.016.000.101:443     https
+172.016.000.101:80      http
+
+git.novatech-llc.com
+172.016.000.102:22      ssh
+172.016.000.102:443     https
+172.016.000.102:80      http
+
+wiki.novatech-llc.com
+172.016.000.103:443     https
+172.016.000.103:80      http
+
+mantis.novatech-llc.com
+172.016.000.104:443     https
+172.016.000.104:80      http
+
+buildsystem.novatech-llc.com
+172.016.000.105:22      ssh
+172.016.000.105:3306    mysql
+172.016.000.105:80      http
+
+testssh.novatech-llc.com
+172.016.000.106:22      ssh
+
+mail.novatech-llc.com
+172.016.000.107:25      smtp
+172.016.000.107:465     smtps
+172.016.000.107:587     smtp+starttls
+
+ldap.novatech-llc.com
+172.016.000.108:389     ldap
+172.016.000.108:443     https
+172.016.000.108:636     ldaps
+172.016.000.108:80      http
+
+buildbot.novatech-llc.com
+172.016.000.110:443     https
+172.016.000.110:80      http
+172.016.000.110:9989    buildbot
+
+nexus.novatech-llc.com
+172.016.000.111:443     https
+172.016.000.111:80      http
+
+frankensteinbs.novatech-llc.com
+172.016.000.120:22      ssh
+172.016.000.120:3306    mysql
+172.016.000.120:69      tftp
+172.016.000.120:80      http
+
+192.168.122.001:53      dns
+
+ssh:22
+    daedalus
+    git
+    buildsystem
+    testssh
+    frankensteinbs
+
+smtp:25
+    mail
+
+tftp:69
+    frankensteinbs
+
+http:80
+    svn
+    git
+    wiki
+    mantis
+    buildsystem
+    ldap
+    buildbot
+    nexus
+    frankensteinbs
+
+ldap:389
+    ldap
+
+https:443
+    svn
+    git
+    wiki
+    mantis
+    buildsystem
+    ldap
+    buildbot
+    nexus
+    frankensteinbs
+
+smtps:465
+    mail
+
+smtp+starttls:587
+    mail
+
+ldaps:636
+    ldap
+
+mysql:3306
+    daedalus
+    buildsystem
+    frankensteinbs
+
+buildbot:9989
+    buildbot
diff --git a/roles/haproxy/files/haproxy.cfg b/roles/haproxy/files/haproxy.cfg
index e44016891..eccf2ffc2 100644
--- a/roles/haproxy/files/haproxy.cfg
+++ b/roles/haproxy/files/haproxy.cfg
@@ -8,17 +8,13 @@ defaults
     timeout connect 5s
     timeout client 50s
     timeout server 50s
+    default-server init-addr last,libc,none
 
 #enable stats page
 listen stats
-    bind :9000
+    bind *:9000
     mode http
     stats enable
-    stats hide-version
-    stats realm Haproxy\ Statistics
-    stats uri /haproxy_stats
-    stats auth stats:stats
-    stats refresh 5s
 
 #
 frontend container_https_in
@@ -26,6 +22,11 @@ frontend container_https_in
     bind *:80
     bind *:443 ssl crt /etc/ssl/private/haproxy.pem
 
+    http-request set-header X-Forwarded-Proto https
+    http-request set-header X-SSL %[ssl_fc]
+
+    http-request redirect scheme https code 301 if !{ ssl_fc }
+
     acl is_buildbot     hdr(host) eq buildbot.novatech-llc.com
     acl is_gitlab       hdr(host) eq git.novatech-llc.com
     acl is_ldapadmin    hdr(host) eq ldap.novatech-llc.com
@@ -36,63 +37,57 @@ frontend container_https_in
 
     use_backend backend-buildbot if is_buildbot
     use_backend backend-nexus if is_nexus
-
-frontend container_ldap_in
-    mode tcp
-    bind *:389
-    bind *:636
-
-    use_backend backend-ldap
-
-frontend container_smtp_in
-    mode tcp
-    bind *:25
-    bind *:465
-    bind *:587
-
-    use_backend backend-exim4
-
-frontend container_ssh_in
-    mode tcp
-    bind *:22
-    default_backend
+    use_backend backend-svn if is_svn
+    use_backend backend-ldapadmin if is_ldapadmin
+    use_backend backend-gitlab if is_gitlab
+    use_backend backend-mantisbt if is_mantisbt
+    use_backend backend-wiki if is_wiki
+
+# frontend container_ldap_in
+#     mode tcp
+#     bind *:389
+#     bind *:636
+#     use_backend backend-ldap
+
+# frontend container_smtp_in
+#     mode tcp
+#     bind *:25
+#     bind *:465
+#     bind *:587
+#     use_backend backend-exim4
+
+# frontend container_ssh_in
+#     mode tcp
+#     bind *:22
+#     default_backend
 
 #
 backend backend-svn
-    redirect scheme https code 301 if !{ ssl_fc }
-    server svn.frontend:80 check
+    server svn1 svn.frontend:80
 
-backend backend-phpldapadmin
-    redirect scheme https code 301 if !{ ssl_fc }
-    server phpldapadmin.frontend:80 check
+backend backend-ldapadmin
+    server phpldapadmin1 phpldapadmin.frontend:80
 
 backend backend-exim4
-    server exim4.frontend:25 check
+    server exim4_1 exim4.frontend:25
 
 backend backend-gitlab
-    redirect scheme https code 301 if !{ ssl_fc }
-    server gitlab.frontend:80 check
+    server gitlab1 gitlab.frontend:80
 
 backend backend-wiki
-    redirect scheme https code 301 if !{ ssl_fc }
-    server wiki.frontend:80 check
+    server wiki1 wiki.frontend:80
 
 backend backend-mantisbt
-    redirect scheme https code 301 if !{ ssl_fc }
-    server mantisbt.frontend:80 check
+    server mantisbt1 mantisbt.frontend:80
 
 backend backend-build
-    redirect scheme https code 301 if !{ ssl_fc }
-    server build.frontend:80 check
+    server build1 build.frontend:80
 
 backend backend-testssh
-    redirect scheme https code 301 if !{ ssl_fc }
-    server testssh.frontend:80 check
+    server testssh1 testssh.frontend:80
 
 backend backend-buildbot
-    redirect scheme https code 301 if !{ ssl_fc }
-    server buildbot.frontend:80 check
+    server buildbot1 buildbot.frontend:80
 
 backend backend-nexus
-    redirect scheme https code 301 if !{ ssl_fc }
-    server nexus.frontend:80 check
+    server nexus1 nexus3.frontend:8081
